GVFS/GVFS.Common/Git/GitProcess.cs

```diff
@@ -505,6 +510,11 @@ public Process GetGitProcess(string command, string workingDirectory, string dot
                     this.gitBinPath,
                     this.gvfsHooksRoot ?? string.Empty);
 
+            if (gitObjectsDirectory != null)
+            {
+                processInfo.EnvironmentVariables["GIT_OBJECT_DIRECTORY"] = gitObjectsDirectory;

```

Comment:
Maybe more of a question for @derrickstolee - how is it decided whether parameters are specified via an environment variable vs command line argument?

Reply:
It seems to be a bit random, but also depends on whether it is a porcelain or plumbing command.

* [`git prune-packed`](https://git-scm.com/docs/git-prune-packed) doesn't have a command-line option.
* [`git unpack-objects`](https://git-scm.com/docs/git-unpack-objects) doesn't have a command-line option.
* [`git commit-graph`](https://git-scm.com/docs/git-commit-graph) has `--object-dir`.
* `git multi-pack-index` has `--object-dir`.